examples/digestauth.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/**
+ * This example shows how to do Digest authentication.
+ * *
+ * @copyright Copyright (C) 2009-2015 fruux GmbH (https://fruux.com/).
+ * @author Markus Staab
+ * @license http://sabre.io/license/ Modified BSD License
+ */
+$userList = [
+    "user" => "password",
+    "user" => "password",
+];
+
+use Sabre\HTTP\Sapi;
+use Sabre\HTTP\Response;
+use Sabre\HTTP\Auth;
+
+// Find the autoloader
+$paths = [
+    __DIR__ . '/../vendor/autoload.php',
+    __DIR__ . '/../../../autoload.php',
+    __DIR__ . '/vendor/autoload.php',
+
+];
+
+foreach ($paths as $path) {
+    if (file_exists($path)) {
+        include $path;
+        break;
+    }
+}
+
+$request = Sapi::getRequest();
+$response = new Response();
+
+$digestAuth = new Auth\Digest("Locked down area", $request, $response);
+$digestAuth->init();
```

Comment:
why does our api require to call `init` before doing the auth?

in the `Basic` auth we dont need that, so I would expect the digest class to auto-init itself when required instead of having this method as part of the public api